### ISSUE TITLE
[CI] Do not run dead links and formatting checks on autogenerated files

### ISSUE THREAD

**Issue by angelikatyborska** [exercism/v3#2547]
TL;DR: Autogenerated files (`languages/README.md`, `reference/README.md`, `reference/stories/README.md`) are not supposed to be edited by anyone but the GHA that generated them, but editing other files (e.g. renaming concept exercises) can cause those autogenerated files to fail the dead links check because it is run on all files. If I'm not supposed to edit autogenerated files, their content shouldn't make my PR checks fail, especially when I did not edit them.

Longer story:

When I did the renaming of all the Elixir exercises (https://github.com/exercism/v3/pull/2544/files), I branched off master when it had some dead links. I did a global replace of all exercise names including those autogenerated files because it would be more difficult to skip them. My PR check failed because of those dead links that were on master.

Somebody fixed them on master in the meantime, so I thought that just rebasing my branch onto master would help, but it didn't. The file `reference/README.md` is formatted in a way that if a single line changes (e.g. a single exercise will be renamed), the WHOLE file is going to change if you reformat it because line lengths have to match, and you do have to reformat it, because if you don't then the check for formatting files will fail. Because the whole file had changes, rebasing my branch onto master just simply overwrote the fixes for dead links that somebody else did.

I thought maybe I can just regenerate the autogenerated files locally on my branch, but the scripts for that are in F# and require dotnet, and I'm on a mac, know nothing about this environment/language so I just couldn't get it to run...

I was forced to do the global replace on all files again, while praying that nobody else changes anything on master before I get the chance to merge my PR because that would bring me back to the problem with changes in `reference/README.md` being overwritten because the whole file gets changed when you reformat it.

**Comment by ErikSchierboom** [comment]
I'm fine with skipping the link checking in those files. It looks like the Markdown link checker we use [doesn't have an ignore option](https://github.com/tcort/markdown-link-check#usage). This means that we should hard-code the exceptions in the [`valid-links.yml` workflow](https://github.com/exercism/v3/blob/master/.github/workflows/valid-links.yml#L40), which looks like this:

```bash
for file in $files; do
  if [ -f "$file" ]
  then
    markdown-link-check -q -c .github/markdown-link-check-config.json "$file"
  fi
done
```
